### PR TITLE
BUG: Add some missing C error handling

### PR DESCRIPTION
**Pull request by eric-wieser**
Mostly irrelevant corner cases, I don't think any of these are testable.

### CODE REVIEW

File: numpy/core/src/multiarray/descriptor.c [hunk 1]
```diff
@@ -1220,7 +1228,14 @@ _convert_from_dict(PyObject *obj, int align)
                     Py_DECREF(tup);
                     goto fail;
                 }
-                PyDict_SetItem(fields, title, tup);
+                else if (PyErr_Occurred()) {
+                    /* MemoryError during dict lookup */
```
Comment (seberg) [inline numpy/core/src/multiarray/descriptor.c]: I wonder if that can actually happen, hehe :) (not that it matters).

File: numpy/core/src/multiarray/descriptor.c [hunk 2]
```diff
@@ -2981,8 +3009,13 @@ PyArray_DescrNewByteorder(PyArray_Descr *self, char newendian)
                 Py_INCREF(old);
                 PyTuple_SET_ITEM(newvalue, i, old);
             }
-            PyDict_SetItem(newfields, key, newvalue);
```
Comment (seberg) [inline numpy/core/src/multiarray/descriptor.c]: You missed `newvalue` above? Not that I care much TBH.

File: numpy/linalg/umath_linalg.c.src [hunk 3]
```diff
@@ -3654,17 +3661,22 @@ PyObject *PyInit__umath_linalg(void)
...
+    if (ret < 0) {
+        return NULL;
+    }
 
     /* Load the ufunc operators into the module's namespace */
-    addUfuncs(d);
-
-    if (PyErr_Occurred()) {
-        PyErr_SetString(PyExc_RuntimeError,
-                        "cannot load _umath_linalg module.");
```
Comment (seberg) [inline numpy/linalg/umath_linalg.c.src]: Could change the error type, but it does not matter really. Also I think that Python replaces/chains it with an ImportError in any case, at least nowadays.

**Review by seberg — APPROVED** [review]
LGTM, none of these are really something likely to actually get triggered, but it is cleaner and best practices. If nobody else has an opinion, I think I will just merge it fairly soon (or someone else just go ahead).

**Comment by mattip** [comment]
Thanks @eric-wieser

**Pull request merged**